mailpoet/lib/AdminPages/Pages/AutomationAnalytics.php

```diff
@@ -0,0 +1,102 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\AdminPages\Pages;
+
+use MailPoet\AdminPages\PageRenderer;
+use MailPoet\Automation\Engine\Mappers\AutomationMapper;
+use MailPoet\Automation\Engine\Registry;
+use MailPoet\Automation\Engine\Storage\AutomationStorage;
+use MailPoet\Form\AssetsController;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoet\WP\Notice as WPNotice;
+
+class AutomationAnalytics {
+
+  /** @var AssetsController */
+  private $assetsController;
+
+  /** @var PageRenderer */
+  private $pageRenderer;
+
+  /** @var AutomationStorage */
+  private $automationStorage;
+
+  /** @var AutomationMapper */
+  private $automationMapper;
+
+  /** @var Registry */
+  private $registry;
+
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    AssetsController $assetsController,
+    PageRenderer $pageRenderer,
+    AutomationStorage $automationStorage,
+    AutomationMapper $automationMapper,
+    Registry $registry,
+    WPFunctions $wp
+  ) {
+    $this->assetsController = $assetsController;
+    $this->pageRenderer = $pageRenderer;
+    $this->automationStorage = $automationStorage;
+    $this->automationMapper = $automationMapper;
+    $this->registry = $registry;
+    $this->wp = $wp;
+  }
+
+  public function render() {
+    $this->assetsController->setupAutomationAnalyticsDependencies();
+
+    $id = isset($_GET['id']) ? (int)$_GET['id'] : null;
+    $automation = $id ? $this->automationStorage->getAutomation($id) : null;
+    if (!$automation) {
+      $notice = new WPNotice(
+        WPNotice::TYPE_ERROR,
+        __('Automation not found.', 'mailpoet')
+      );
+      $notice->displayWPNotice();
+      $this->pageRenderer->displayPage('blank.html');
+      return;
+    }
+
+    $this->pageRenderer->displayPage('automation/analytics.html', [
+      'registry' => $this->buildRegistry(),
+      'context' => $this->buildContext(),
+      'automation' => $this->automationMapper->buildAutomation($automation),
+      'sub_menu' => 'mailpoet-automation',
```

Comment:
When you rebase on `trunk`, please remove `sub_menu`. This is no longer needed since https://github.com/mailpoet/mailpoet/pull/4933/files.

Reply:
Addressed in a183e254f